addon/@ember/test-waiters/wait-for.ts

```diff
@@ -96,3 +135,33 @@ function isThenable(
     typeof maybePromise.then === 'function'
   );
 }
+
+function isGenerator(
+  maybeGenerator: any | CoroutineGenerator<unknown>
+): maybeGenerator is CoroutineGenerator<unknown> {
+  return typeof maybeGenerator[Symbol.iterator] === 'function';
+}
+
+const GENERATOR_WAITER = buildWaiter('@ember/test-waiters:generator-waiter');
+
+function waitForGenerator<T>(
```

Comment:
Lets rename this to `waitForIterator` (generator is the function which returns an iterator, right?)...

Reply:
I believe this naming is accurate -- see above

Reply:
Hmmm I think I answered my own question. The generator function returns a generator object. 👍

Reply:
We agreed this is correct, as we're specifically targeting generators.